### PR TITLE
Support GEMS MASTER for AmoebaNet and ResNet 

### PR DESCRIPTION
**Pull request by RadhaGulhane13**

### CODE REVIEW

File: benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+    def writelines(self, datas):
+        self.stream.writelines(datas)
+        self.stream.flush()
+
+    def __getattr__(self, attr):
+        return getattr(self.stream, attr)
+
+
+def init_processes(backend="tcp"):
```
Comment (Quentin-Anthony) [inline benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py:41]: should be `backend="mpi"`

File: benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+        root=datapath, train=True, download=True, transform=transform
+    )
+    my_dataloader = torch.utils.data.DataLoader(
+        trainset,
+        batch_size=times * batch_size,
+        shuffle=False,
+        num_workers=0,
+        pin_memory=True,
+    )
+    size_dataset = 50000
```
Comment (Quentin-Anthony) [inline benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py:218]: why is this hardcoded?

File: benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+        num_classes=num_classes,
+        transform=transform,
+        target_transform=None,
+        random_offset=0,
+    )
+    my_dataloader = torch.utils.data.DataLoader(
+        my_dataset,
+        batch_size=batch_size * times,
+        shuffle=False,
+        num_workers=0,
```
Comment (Quentin-Anthony) [inline benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py:232]: expose `num_workers` as an arg unless we have a good reason to force it here

File: benchmarks/gems_master_model/benchmark_resnet_gems_master.py [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+    def writelines(self, datas):
+        self.stream.writelines(datas)
+        self.stream.flush()
+
+    def __getattr__(self, attr):
+        return getattr(self.stream, attr)
+
+
+def init_processes(backend="tcp"):
```
Comment (Quentin-Anthony) [inline benchmarks/gems_master_model/benchmark_resnet_gems_master.py:41]: Should be `backend="mpi"`

File: benchmarks/gems_master_model/benchmark_resnet_gems_master.py [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+    trainset = torchvision.datasets.ImageFolder(
+        datapath,
+        transform=transform,
+        target_transform=None,
+    )
+    my_dataloader = torch.utils.data.DataLoader(
+        trainset,
+        batch_size=times * batch_size,
+        shuffle=True,
+        num_workers=0,
```
Comment (Quentin-Anthony) [inline benchmarks/gems_master_model/benchmark_resnet_gems_master.py:211]: same arg comment here

File: benchmarks/gems_master_model/benchmark_resnet_gems_master.py [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+        num_classes=num_classes,
+        transform=transform,
+        target_transform=None,
+        random_offset=0,
+    )
+    my_dataloader = torch.utils.data.DataLoader(
+        my_dataset,
+        batch_size=batch_size * times,
+        shuffle=False,
+        num_workers=0,
```
Comment (Quentin-Anthony) [inline benchmarks/gems_master_model/benchmark_resnet_gems_master.py:240]: same arg comment here

File: benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py (hunk 3 above)
Comment (RadhaGulhane13) [reply benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py:232]: Are you suggesting removing num_workers from the argument list or using it as a command-line argument?

File: benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py (hunk 1 above)
Comment (RadhaGulhane13) [reply benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py]: done all changes related to "tcp"->"mpi"

File: benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py (hunk 3 above)
Comment (Quentin-Anthony) [reply benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py]: Don't remove `num_workers` as an arg. It should still be a function arg and default to zero, but I want the user to be able to pass it in the benchmark `args` similar to `args.split_size` https://github.com/OSU-Nowlab/MPI4DL/pull/11/files#diff-68cd23c73cb37b7f0b9ae6cdde9b4d13819b4461ac67588399977b94cc1f528bR65

Comment (RadhaGulhane13) [reply benchmarks/gems_master_model/benchmark_amoebanet_gems_master.py]: Done.

**Review by Quentin-Anthony — APPROVED** [review]

**Pull request merged**